### ISSUE TITLE
Create angular environment

### ISSUE THREAD

**Issue by remcohaszing** [Gillespie59/eslint-plugin-angular#331]
It would be nice to create an `angular` environment. This would set the global `angular` and the globals for `angular-mocks`.

This is an [accepted feature](https://github.com/eslint/eslint/issues/4782) with no set milestone yet. I think this would be very useful to this plugin and should be implemented as soon as that feature is implemented.

This would be useful to prevent issue reports similar to #330.

**Comment by Gillespie59** [comment]
So we should add this configuration in our index.js ? 

```javascript
module.exports = {
    rules: {},
    environments: {
        custom: {
            globals: { 
                    "angular": true ,
                    "module": true ,
                    "inject": true 
            }
        }
    }
};
```

**Comment by remcohaszing** [comment]
I suppose this should cover the basics:
```
module.exports = {
  rules: {},
  environments: {
    angular: {
      globals: {
        angular: true
      }
    },
    // https://docs.angularjs.org/api/ngMock
    inject: {
      globals: {
        angular: true,
        inject: true,
        module: true
      }
    },
    // http://www.protractortest.org/#/api
    protractor: {
      globals: {
        element: true,
        $: true,
        $$: true,
        browser: true,
        by: true,
        protractor: true
      }
    }
  }
};
```
I didn't really look into it in that much detail, haven't tried it manually yet to be honest. I didn't expect a sudden flood of git patches. ;)

I'm not sure if it's recommended by the ESLint team to inherit configurations. E.g. Protractor always runs in NodeJS. Should it include `module`, `exports` and `require`?
